app/lib/capacitor/index.js

```diff
@@ -0,0 +1,142 @@
+const log = require('../helpers/logger')('app:capacitor'),
+  warn = require('../helpers/logger')('app:capacitor', 'red'),
+  fse = require('fs-extra'),
+  path = require('path'),
+  { spawn } = require('../helpers/spawn'),
+  getCapacitorBinaryBath = require('../capacitor/getCapacitorBinaryPath'),
+  appPaths = require('../app-paths')
+
+class CapacitorRunner {
+  run(quasarConfig) {
+    return new Promise(async resolve => {
+      const cfg = quasarConfig.getBuildConfig(),
+        url = cfg.build.APP_URL,
+        platform = cfg.ctx.targetName
+      // Make sure there is an index.html, otherwise Capacitor will crash
+      fse.ensureFileSync(appPaths.resolve.app('./dist/capacitor/index.html'))
+      // Copy app data
+      await this.__runCapacitorCommand(['sync', cfg.ctx.targetName])
+      this.__setCapacitorConfig(platform, url)
+      // Make sure cleartext is enabled
+      if (cfg.ctx.targetName === 'android') {
+        const androidManifest = fse.readFileSync(
+          appPaths.resolve.capacitorAndroid('app/src/main/AndroidManifest.xml'),
+          'utf8'
+        )
+        if (!androidManifest.match('android:usesCleartextTraffic="true"')) {
+          warn(
+            'Cleartext must be enabled to connect to the dev server on Android!'
+          )
+          warn(
+            'Add `android:usesCleartextTraffic="true"` to android/app/src/main/AndroidManifest.xml in the `application` tag'
+          )
+          // TODO: actual url with instructions
+          warn('See [need url] for more instructions')
+          process.exit(1)
+        }
+      }
+      await this.__runCapacitorCommand(['open', cfg.ctx.targetName])
+      if (!/^192/.test(url)) {
```

Comment:
I believe this is not always true.. Quasar picked up my local network IP as url... I dont know

Reply:
@rstoenescu what is the best way to test if the Quasar dev server url is on the network or just a localhost url?